Services/TMS/TableRelations/classes/class.SelectableReportTableGUI.php

```diff
@@ -8,18 +12,93 @@ class SelectableReportTableGUI extends ilTable2GUI {
 	protected $order = [];
 	protected $selectable = [];
 	protected $internal_sorting_columns = [];
+	protected $export_writer = [];
 
-	public function __construct($a_parent_gui, $a_cmd) {
-		global $ilCtrl;
 
+	public function __construct($a_parent_gui, $a_cmd) {
+		global $DIC;
+		$g_ctrl = $DIC['ilCtrl'];
+		$this->setId("elpt_".$a_parent_gui->id);
 		parent::__construct($a_parent_gui, $a_cmd);
 		$this->setEnableTitle(false);
 		$this->setTopCommands(false);
 		$this->setEnableHeader(true);
-		$this->setFormAction($ilCtrl->getFormAction($a_parent_gui,$a_cmd));
+		$this->setFormAction($g_ctrl->getFormAction($a_parent_gui,$a_cmd));
 		$this->columns_determined = false;
-		$this->setId("report_table");
 		$this->setExternalSorting(true);
+		$this->export_formats = [];
+	}
+
+	public function addExporter(
+		ExcelWrapper\Writer $writer,
+		$format_id,
+		$export_format_title,
+		$export_format_mine)
+	{
+		assert('is_int($format_id)');
+		assert('is_string($export_format_title)');
+		assert('is_string($export_format_mine)');
+		$this->export_formats[$format_id] = $export_format_title;
+		$this->export_writer[$format_id] = $writer;
+		$this->export_mime[$format_id] = $export_format_mine;
+	}
+
+	public function exportData($format_id, $send = false)
```

Comment:
Ich finde leider nirgendwo in den 3 PR einen Aufruf von der Funktion hier.
Vergessen oder Absicht?

Reply:
geerbt von ilTable2GUI ...